### DISCUSSION TITLE
Unable to call component method. Public method not found on component.

### DISCUSSION THREAD

**Discussion by habibmhamadi** [livewire/livewire#3282] [1. Help] [answered]
I have a CRUD livewire component. the component lists data and there are blade dialogs for creating, deleting, and editing that is conditionally included in the livewire component. everything works fine except when I create a new record and then try to edit or delete this new record (should open edit/delete dialog by calling a method) but it gives me the error.
It works on other existing records, the error appears on the newly created record. but it works when I manually refresh the page.

Livewire component:
![handler](https://user-images.githubusercontent.com/34987353/125159721-57d34000-e18e-11eb-9479-85d9d55a9522.png)

Livewire view:
![view](https://user-images.githubusercontent.com/34987353/125159729-67528900-e18e-11eb-85c0-603fabe17181.png)

Livewire\Exceptions\MethodNotFoundException
Unable to call component method. Public method [prepareEdit({] not found on component: [category.index]
http://localhost:8000/admin/category

**Comment by joshhanley** [comment, marked as answer]
@habibmhamadi try pass the id into the method call `prepareEdit({{ $category->id ]})` instead of the whole object.

Also please don't paste screenshots of code, instead actually paste your code as text and put in triple backticks \``` so we are able to select your code and change pieces of it, to make it easier to assist you.

Hope this helps!

**Reply by habibmhamadi** [reply]
Thank you it worked with id.
Passing the whole object works on Windows (PHP v8) but not on Ubuntu (PHP v7.4).
